server/src/main/java/org/elasticsearch/action/support/single/shard/TransportSingleShardAction.java

```diff
@@ -118,8 +119,18 @@ protected void doRun() throws Exception {
             }
         });
     }
+
+    @Deprecated
     protected abstract Response newResponse();
 
+    protected Writeable.Reader<Response> getResponseReader() {
```

Comment:
thanks for this @romseygeek!
I tried my hand at a follow-up to remove the adapter for this here #41985.